packages/components/src/test/table/UsePresentationTable.test.tsx

```diff
@@ -171,4 +171,101 @@ describe("usePresentationTableWithUnifiedSelection", () => {
+    const keys = ["this is not a valid key", ""];
+    const { result } = renderHook(() => usePresentationTableWithUnifiedSelection(initialProps), { wrapper: Wrapper });
+    const onSelectCallback = result.current.onSelect;
+
+    const replaceSpy = sinon.stub(Presentation.selection, "replaceSelection");
+    onSelectCallback(keys);
+
+    expect(replaceSpy).to.have.been.calledOnceWithExactly("UnifiedSelectionContext", {}, [], 1);
+  });
+
+  it("Gets valid keys for rows that are not loaded and does not pass any to the selection with onSelect", async () => {
+    const stringifiedKeys = [createTestECInstanceKey()].map((key) => JSON.stringify(key));
+
+    const { result } = renderHook(() => usePresentationTableWithUnifiedSelection(initialProps), { wrapper: Wrapper });
+    const onSelectCallback = result.current.onSelect;
+
+    const replaceSpy = sinon.stub(Presentation.selection, "replaceSelection");
+    onSelectCallback(stringifiedKeys);
+
+    expect(replaceSpy).to.have.been.calledOnceWithExactly("UnifiedSelectionContext", {}, [], 1);
+  });
+
+  it("returns an array of selectedRows when keys are passed correctly on selectionChange event", async () => {
+    const keys = new KeySet([createTestECInstanceKey()]);
+    sinon.stub(Presentation.selection, "getSelection").returns(new KeySet([createTestECInstanceKey()]));
+
+    const { descriptor, item } = getMockData();
+    presentationManagerMock.setup(async (x) => x.getContentDescriptor(moq.It.is((options) => options.keys.size === keys.size))).returns(async () => descriptor);
+    presentationManagerMock
+      .setup(async (x) => x.getContentAndSize(moq.It.is((options) => options.keys.size === keys.size)))
+      .returns(async () => ({ content: new Content(descriptor, [item]), size: 1 }));
+
+    const { result } = renderHook(() => usePresentationTableWithUnifiedSelection(initialProps), { wrapper: Wrapper });
+
+    await waitFor(() => expect(result.current.isLoading).to.be.false);
+    Presentation.selection.addToSelection("UnifiedSelectionContext", initialProps.imodel, keys, 1);
+
+    await waitFor(() => expect(result.current.isLoading).to.be.false);
+    const resultAfterAdding = result.current.selectedRows;
+    expect(resultAfterAdding.length).to.be.equal(1);
+  });
+
+  it("returns an empty array of selectedRows when keys are passed from the wrong level on selectionChange event", async () => {
+    const { result } = renderHook(() => usePresentationTableWithUnifiedSelection(initialProps), { wrapper: Wrapper });
+
+    await waitFor(() => expect(result.current.isLoading).to.be.false);
+    Presentation.selection.addToSelection("UnifiedSelectionContext", initialProps.imodel, new KeySet([createTestECInstanceKey()]), 3);
+
+    await waitFor(() => expect(result.current.isLoading).to.be.false);
+    const resultAfterAdding = result.current.selectedRows;
+    expect(resultAfterAdding.length).to.be.equal(0);
+  });
+
+  function getMockData(): { descriptor: Descriptor; item: Item } {
```

Comment:
I believe this function could also set up `presentationManagerMock` to further reduce clutter in the tests

Reply:
Could you also check existing tests and see if they could reuse this function? E.g. it looks like `loads columns and rows with keys from unified selection` could...

Reply:
I did check on this before, but decided to leave them untouched since these tests do explicit checking of the values in the setting up part. I think getting the values that are checked from a separate call would reduce the readability of the test.